src/screens/notification/container/notificationContainer.js

```diff
@@ -168,7 +169,9 @@ class NotificationContainer extends Component {
     await this.setState({ selectedFilter: value, endOfNotification: false, selectedIndex: ind });
   };
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
+  // Note: This snippet is causing double api call which cause scroll position reset
+  /*
+    UNSAFE_componentWillReceiveProps(nextProps) {
     const { selectedFilter } = this.state;
     const { currentAccount } = this.props;
 
```

Comment:
@aliseyalvi the logic is checking if user account changes, refetch the comments... I think the logic is not setup write, check how there is an OR statement with logic just checking if current tab is Notifications tab and user is logged in... which means, activities will be refetched on every prop update...

The logic should simply by... if account change then refresh... no need to check tab or anything else.....because even if tab is not notification we will still have to fetch new notifications irrespective...

long story short....try
`if(currentAccount && nextProps.currentAccount && nextProps.currentAccount.name !== currentAccount.name)`

p.s. make sure to test account change and logging out as well.

Reply:
@noumantahir added condition for account change and test as well. working good now. check update here https://github.com/ecency/ecency-mobile/pull/2212/commits/d7034518003562f521981ed06ed8f5ec2ae5f8d7